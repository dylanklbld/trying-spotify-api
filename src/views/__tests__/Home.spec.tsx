import { fireEvent, render, screen } from '@testing-library/react';

import HomePage from '../Home';
import React from 'react';

describe('HomePage', () => {
  it('should render HomePage', () => {
    const { container } = render(<HomePage />)

    expect(container).toMatchSnapshot()
  })

  it('should redirect to Spotify', () => {
    render(<HomePage />)
    const url = "http://dummy.com";
    global.window = Object.create(window);
    Object.defineProperty(window, 'location', {
      value: {
        href: url
      }
    });

    const loginToSpotifyButton = screen.getByText("Login to spotify")
    fireEvent.click(loginToSpotifyButton)

    expect(window.location.href).toContain('https://accounts.spotify.com/authorize');
  })
})