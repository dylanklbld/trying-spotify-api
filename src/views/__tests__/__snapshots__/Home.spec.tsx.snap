// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should render HomePage 1`] = `
<div>
  <div
    class="login"
  >
    <button
      class="sc-bdfBwQ brfkKw"
      type="submit"
    >
      Login to spotify
    </button>
  </div>
</div>
`;
